python/hail/table.py

```diff
@@ -2349,6 +2368,8 @@ def distinct(self) -> 'Table':
         -------
         :class:`.Table`
         """
+        if self.key is None:
```

Comment:
Where would be a good place for that? If I put it in `hail.methods.misc` with the other `require_...` methods, I can't use it in `table.py` because of circular imports.

Reply:
try removing the import from misc and making the usage `hl.Table`

Reply:
Thanks, got it.

Reply:
This is my fault for not including a sensible organization.Python does make it difficult, though.

Reply:
especially with type annotations! those are the killer, because they need to be evaluated in the module scope rather than the function scope (evaluated later)